src/Avalonia.FreeDesktop/DBusIme/IBus/IBusX11TextInputMethod.cs

```diff
@@ -49,8 +46,7 @@ private void OnForwardKey((uint keyval, uint keycode, uint state) k)
             });
         }
 
-        
-        private void OnCommitText(object wtf)
+        private void OnCommitText(Exception? e, object wtf)
```

Comment:
I wonder if we could remove the reflection usage here. SRE-powered deserializer passes some dynamically resolved tuple type here.

Reply:
Since per xml definition the type is a variant, the tuple will always be dynamically resolved. Though perhaps it's possible to use type matching with `is ValueTuple<..., ..., string>`?

Reply:
Qt has `QDBusArgument`  that allows to manually encode/decode variants. Same with GTK and GVariant/GVariantBuilder.

Reply:
I fail to see how this would solve the problem, but perhaps it's just me missing something obvious.
As I understand it, we'd still have to know the clr types in order to read them / parse the variant's signature manually, which is exactly what is done by Tmds.DBus.Protocol.

Reply:
The variant signature can contain a fixed set of primitive types that could be deserialized into `DBusVariant` container type or something. Just like Qt and GTK do.

Reply:
A variant can, in theory, contain custom structs whose layout can only be known at runtime through the signature, so the only real option I see here is type pattern matching.

Reply:
A variant type can always be decoded into something like:
```
    public class DBusBasicTypeItem : DBusItem
    {
        public object Value { get; }

        public DBusBasicTypeItem(byte b)
        {
            Value = b;
        }
        /// Ctors for all other basic types here
    }

    public class DBusArray : DBusItem //, IList<DBusItem>
    {
        // list implementation
    }

    public class DBusStruct : DBusItem //, IList<DBusItem>
    {
        // list implementation
    }

    public class DBusDictEntry : DBusItem
    {
        public DBusBasicTypeItem Key { get; set; }
        public DBusItem Value { get; set; }
    }
```

That's how Qt and GTK handle dbus variants.